java/src/main/java/com/powsybl/python/network/NetworkModificationCFunctions.java

```diff
@@ -0,0 +1,150 @@
+ */
+package com.powsybl.python.network;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.iidm.network.*;
+import com.powsybl.iidm.network.extensions.ConnectablePosition;
+import com.powsybl.python.commons.CTypeUtil;
+import com.powsybl.python.commons.Directives;
+import com.powsybl.python.commons.PyPowsyblApiHeader;
+import org.apache.commons.lang3.Range;
+import org.graalvm.nativeimage.IsolateThread;
+import org.graalvm.nativeimage.ObjectHandle;
+import org.graalvm.nativeimage.ObjectHandles;
+import org.graalvm.nativeimage.c.CContext;
+import org.graalvm.nativeimage.c.function.CEntryPoint;
+import org.graalvm.nativeimage.c.type.CCharPointer;
+import org.graalvm.nativeimage.c.type.CIntPointer;
+
+import java.util.*;
+
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.getUnusedOrderPositionsAfter;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.getUnusedOrderPositionsBefore;
+import static com.powsybl.python.commons.Util.createIntegerArray;
+import static com.powsybl.python.commons.Util.doCatch;
+
+/**
+ * @author Coline Piloquet <coline.piloquet at rte-france.com>
+ */
+@CContext(Directives.class)
+public final class NetworkModificationCFunctions {
+
+    private NetworkModificationCFunctions() {
+    }
+
+    @CEntryPoint(name = "getConnectablesOrderPositions")
+    public static PyPowsyblApiHeader.ArrayPointer<PyPowsyblApiHeader.SeriesPointer> getConnectablesOrderPositions(IsolateThread thread, ObjectHandle networkHandle,
+                                                     CCharPointer voltageLevelId, PyPowsyblApiHeader.ExceptionHandlerPointer exceptionHandlerPtr) {
+        return doCatch(exceptionHandlerPtr, () -> {
+            String voltageLevelIdStr = CTypeUtil.toString(voltageLevelId);
+            Network network = ObjectHandles.getGlobal().get(networkHandle);
+
+            VoltageLevel voltageLevel = network.getVoltageLevel(voltageLevelIdStr);
+            Map<String, List<ConnectablePosition.Feeder>> feederPositionsOrders = getFeedersByConnectable(voltageLevel);
+            return Dataframes.createCDataframe(Dataframes.feederMapMapper(), feederPositionsOrders);
+        });
+
+    }
+
+    @CEntryPoint(name = "getUnusedConnectableOrderPositionsBefore")
+    public static PyPowsyblApiHeader.ArrayPointer<CIntPointer> getUnusedConnectableOrderPositionsBefore(IsolateThread thread, ObjectHandle networkHandle,
+                                                                                             CCharPointer busbarSectionId,
+                                                                                             PyPowsyblApiHeader.ExceptionHandlerPointer exceptionHandlerPtr) {
+        Network network = ObjectHandles.getGlobal().get(networkHandle);
+        String busbarSectionIdStr = CTypeUtil.toString(busbarSectionId);
+        BusbarSection busbarSection = network.getBusbarSection(busbarSectionIdStr);
+        try {
```

Comment:
(if we keep the method)
The try catch is not useful, but we need to wrap the method in a `doCatch` to propagate exceptions to C code.